x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/optimizer/Optimizer.java

```diff
@@ -165,6 +167,9 @@ public LogicalPlan optimize(LogicalPlan verified) {
                 new PushDownAndCombineFilters()
         );
 
+        Batch pushDownOrderBy = new Batch("Push Down Order By",
+            new PushDownAndCombineOrderBy());
```

Comment:
Needs to run isolated from `PushDownAndCombineFilters` in order to avoid interfering push downs.

Reply:
I think the optimizer would never converge to a stable plan in some cases as `OrderBy` and `Filter` would alternately be pushed down.